File: chazm-model/src/main/java/io/github/runtimemodels/notification/DefaultSubscriber.java
package io.github.runtimemodels.notification;

import io.github.runtimemodels.chazm.event.*;
import io.github.runtimemodels.message.L;
import org.slf4j.Logger;

import javax.validation.constraints.NotNull;

class DefaultSubscriber implements Subscriber {

    private static final Logger log = org.slf4j.LoggerFactory.getLogger(DefaultSubscriber.class);

    @Subscribe
    public void event(@NotNull final AchievesEvent event) {
        log.debug(L.RECEIVED_ROLE_GOAL.get(), event.getClass().getSimpleName(), event.getCategory(), event.getRoleId(), event.getGoalId());
    }

    @Subscribe
    public void event(@NotNull final AgentEvent event) {
        log.debug(L.RECEIVED_ID.get(), event.getClass().getSimpleName(), event.getCategory(), event.getId());
    }

    @Subscribe
    public void event(@NotNull final AssignmentEvent event) {
        log.debug(L.RECEIVED_AGENT_ROLE_GOAL.get(), event.getClass().getSimpleName(), event.getCategory(), event.getAgentId(), event.getRoleId(),
                event.getGoalId());
    }

    @Subscribe
    public void event(@NotNull final AttributeEvent event) {
        log.debug(L.RECEIVED_ID.get(), event.getClass().getSimpleName(), event.getCategory(), event.getId());
    }

    @Subscribe
    public void event(@NotNull final CapabilityEvent event) {
        log.debug(L.RECEIVED_ID.get(), event.getClass().getSimpleName(), event.getCategory(), event.getId());
    }

    @Subscribe
    public void event(@NotNull final CharacteristicEvent event) {
        log.debug(L.RECEIVED_ID.get(), event.getClass().getSimpleName(), event.getCategory(), event.getId());
    }

    @Subscribe
    public void event(@NotNull final ContainsEvent event) {
        log.debug(L.RECEIVED_ROLE_CHARACTERISTIC_VALUE.get(), event.getClass().getSimpleName(), event.getCategory(), event.getRoleId(),
                event.getCharacteristicId(), event.getValue());
    }

    @Subscribe
    public void event(@NotNull final HasEvent event) {
        log.debug(L.RECEIVED_AGENT_ATTRIBUTE_VALUE.get(), event.getClass().getSimpleName(), event.getCategory(), event.getAgentId(), event.getAttributeId(),
                event.getValue());
    }

    @Subscribe
    public void event(@NotNull final InstanceGoalEvent event) {
        log.debug(L.RECEIVED_ID.get(), event.getClass().getSimpleName(), event.getCategory(), event.getId());
    }

    @Subscribe
    public void event(@NotNull final ModeratesEvent event) {
        log.debug(L.RECEIVED_PMF_ATTRIBUTE.get(), event.getClass().getSimpleName(), event.getCategory(), event.getPmfId(), event.getAttributeId());
    }

    @Subscribe
    public void event(@NotNull final NeedsEvent event) {
        log.debug(L.RECEIVED_ROLE_ATTRIBUTE.get(), event.getClass().getSimpleName(), event.getCategory(), event.getRoleId(), event.getAttributeId());
    }

    @Subscribe
    public void event(@NotNull final PmfEvent event) {
        log.debug(L.RECEIVED_ID.get(), event.getClass().getSimpleName(), event.getCategory(), event.getId());
    }

    @Subscribe
    public void event(@NotNull final PolicyEvent event) {
        log.debug(L.RECEIVED_ID.get(), event.getClass().getSimpleName(), event.getCategory(), event.getId());
    }

    @Subscribe
    public void event(@NotNull final PossessesEvent event) {
        log.debug(L.RECEIVED_AGENT_CAPABILITY_SCORE.get(), event.getClass().getSimpleName(), event.getCategory(), event.getAgentId(),
                event.getCapabilityId(), event.getScore());
    }

    @Subscribe
    public void event(@NotNull final RequiresEvent event) {
        log.debug(L.RECEIVED_ROLE_CAPABILITY.get(), event.getClass().getSimpleName(), event.getCategory(), event.getRoleId(), event.getCapabilityId());
    }

    @Subscribe
    public void event(@NotNull final RoleEvent event) {
        log.debug(L.RECEIVED_ID.get(), event.getClass().getSimpleName(), event.getCategory(), event.getId());
    }

    @Subscribe
    public void event(@NotNull final SpecificationGoalEvent event) {
        log.debug(L.RECEIVED_ID.get(), event.getClass().getSimpleName(), event.getCategory(), event.getId());
    }

    @Subscribe
    public void event(@NotNull final UsesEvent event) {
        log.debug(L.RECEIVED_ROLE_PMF.get(), event.getClass().getSimpleName(), event.getCategory(), event.getRoleId(), event.getPmfId());
    }

}

File: chazm-model/src/main/java/runtimemodels/chazm/model/notification/DefaultSubscriber.kt
package runtimemodels.chazm.model.notification

import runtimemodels.chazm.model.event.*
import runtimemodels.chazm.model.message.L
import javax.validation.constraints.NotNull

internal open class DefaultSubscriber : Subscriber {

    @Subscribe
    fun event(@NotNull event: AchievesEvent) {
        log.debug(L.RECEIVED_ROLE_GOAL.get(), event.javaClass.simpleName, event.category, event.roleId, event.goalId)
    }

    @Subscribe
    fun event(@NotNull event: AgentEvent) {
        log.debug(L.RECEIVED_ID.get(), event.javaClass.simpleName, event.category, event.id)
    }

    @Subscribe
    fun event(@NotNull event: AssignmentEvent) {
        log.debug(L.RECEIVED_AGENT_ROLE_GOAL.get(), event.javaClass.simpleName, event.category, event.agentId, event.roleId,
            event.goalId)
    }

    @Subscribe
    fun event(@NotNull event: AttributeEvent) {
        log.debug(L.RECEIVED_ID.get(), event.javaClass.simpleName, event.category, event.id)
    }

    @Subscribe
    fun event(@NotNull event: CapabilityEvent) {
        log.debug(L.RECEIVED_ID.get(), event.javaClass.simpleName, event.category, event.id)
    }

    @Subscribe
    fun event(@NotNull event: CharacteristicEvent) {
        log.debug(L.RECEIVED_ID.get(), event.javaClass.simpleName, event.category, event.id)
    }

    @Subscribe
    fun event(@NotNull event: ContainsEvent) {
        log.debug(L.RECEIVED_ROLE_CHARACTERISTIC_VALUE.get(), event.javaClass.simpleName, event.category, event.roleId,
            event.characteristicId, event.value)
    }

    @Subscribe
    fun event(@NotNull event: HasEvent) {
        log.debug(L.RECEIVED_AGENT_ATTRIBUTE_VALUE.get(), event.javaClass.simpleName, event.category, event.agentId, event.attributeId,
            event.value)
    }

    @Subscribe
    fun event(@NotNull event: InstanceGoalEvent) {
        log.debug(L.RECEIVED_ID.get(), event.javaClass.simpleName, event.category, event.id)
    }

    @Subscribe
    fun event(@NotNull event: ModeratesEvent) {
        log.debug(L.RECEIVED_PMF_ATTRIBUTE.get(), event.javaClass.simpleName, event.category, event.pmfId, event.attributeId)
    }

    @Subscribe
    fun event(@NotNull event: NeedsEvent) {
        log.debug(L.RECEIVED_ROLE_ATTRIBUTE.get(), event.javaClass.simpleName, event.category, event.roleId, event.attributeId)
    }

    @Subscribe
    fun event(@NotNull event: PmfEvent) {
        log.debug(L.RECEIVED_ID.get(), event.javaClass.simpleName, event.category, event.id)
    }

    @Subscribe
    fun event(@NotNull event: PolicyEvent) {
        log.debug(L.RECEIVED_ID.get(), event.javaClass.simpleName, event.category, event.id)
    }

    @Subscribe
    fun event(@NotNull event: PossessesEvent) {
        log.debug(L.RECEIVED_AGENT_CAPABILITY_SCORE.get(), event.javaClass.simpleName, event.category, event.agentId,
            event.capabilityId, event.score)
    }

    @Subscribe
    fun event(@NotNull event: RequiresEvent) {
        log.debug(L.RECEIVED_ROLE_CAPABILITY.get(), event.javaClass.simpleName, event.category, event.roleId, event.capabilityId)
    }

    @Subscribe
    fun event(@NotNull event: RoleEvent) {
        log.debug(L.RECEIVED_ID.get(), event.javaClass.simpleName, event.category, event.id)
    }

    @Subscribe
    fun event(@NotNull event: SpecificationGoalEvent) {
        log.debug(L.RECEIVED_ID.get(), event.javaClass.simpleName, event.category, event.id)
    }

    @Subscribe
    fun event(@NotNull event: UsesEvent) {
        log.debug(L.RECEIVED_ROLE_PMF.get(), event.javaClass.simpleName, event.category, event.roleId, event.pmfId)
    }

    companion object {
        private val log = org.slf4j.LoggerFactory.getLogger(DefaultSubscriber::class.java)
    }

}
